tests/parity_client/mod.rs

```diff
@@ -214,11 +218,11 @@ impl ParityClient {
             .wait()
     }
 
-    pub fn deploy_htlc(&self, data: Bytes, value: U256) -> H256 {
+    pub fn deploy_htlc(&self, data: Bytes, value: U256, gas_limit: U256) -> H256 {
         self.sign_and_send(|nonce, gas_price| UnsignedTransaction {
             nonce,
             gas_price,
-            gas_limit: U256::from(500_000),
```

Comment:
This is the other, worse, issue where we hardcode a gas limit of 500,000 when deploying the htlc instead of using our own recommended gas limit.